src/main/java/design/MazeDesignerPresenter.java

```diff
@@ -0,0 +1,65 @@
+package design;
+
+import entities.DesignableMaze;
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.geometry.Pos;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.layout.GridPane;
+import javafx.stage.Popup;
+import javafx.stage.Stage;
+
+public class MazeDesignerPresenter implements MazeDesignerOutputBoundary{
+    public void updateMazeUI(MazeDesignerController md, Button[][] buttonarray){
```

Comment:
make sure not to use controller here, instead pass a request model for example to represent the maze state so you can access the characters in the spaces without the controller